### PR TITLE
Add tests 

### PR DESCRIPTION
**Pull request by JohanRochet**
✅ I have read the [contributing file](https://github.com/NMF-earth/nmf-app/blob/main/contributing.md)

## Summary
Hello, I'm a student in Computer Science and one of my lecturers charge me of choosing an open source application and just writing test on it. NMF-earth is the app I chose.

That's why I went a bit through the code and notice there might be some missing tests, specifically in the utils files. Here's why I wrote new unit tests with the assistance of Stryker, the mutation testing framework. Finally, I wanted to made tests on the user interaction, and so I designed one test for it in the screens. 

This pull request is then essentially some test added to the app. Don't hesitate to contact me for any question and excuse me if I made some _good_ _practice_ mistakes, it's the first time I program with react-native 😉.

### CODE REVIEW

**Comment by PierreBresson** [comment]
Hello @JohanRochet 👋
Thanks for having a look :) Could you use jest instead of Stryker so we don't add any extra dependency?

**Comment by JohanRochet** [comment]
I just deleted the dependency with stryker, unfortunately the interaction tests I wrote in MyLocationScreen.test.tsx require the use of React Native Testing Library which as far as I know can not be replaced by Jest. Do you want me to delete those tests or is it ok for you ?

File: app/screens/MyLocation/__tests__/MyLocationScreen.tests.tsx [hunk 1]
```diff
@@ -10,3 +12,27 @@ it("MyLocationScreen renders correctly", () => {
   const tree = create(<MyLocationScreen />).toJSON();
   expect(tree).toMatchSnapshot();
 });
+
+it("Change current location to ireland", () => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (PierreBresson) [inline app/screens/MyLocation/__tests__/MyLocationScreen.tests.tsx:17]: `import * as redux from 'react-redux'`

Then `const useSelectorMock = jest.spyOn(redux, "useSelector");` so you don't need the `// eslint-disable-next-line @typescript-eslint/no-var-requires`

File: app/screens/MyLocation/__tests__/MyLocationScreen.tests.tsx [hunk 2]
```diff
@@ -10,3 +12,27 @@ it("MyLocationScreen renders correctly", () => {
...
+  const useDispatchMock = jest.spyOn(require("react-redux"), "useDispatch");
+
+  useSelectorMock.mockReturnValue(ElectricityType.france);
+
+  const dummyDispatch = jest.fn();
+  useDispatchMock.mockReturnValue(dummyDispatch);
+
+  render(<MyLocationScreen />);
+  fireEvent.press(screen.root.findByProps({ title: ElectricityType.ireland.toUpperCase() }));
```
Comment (PierreBresson) [inline app/screens/MyLocation/__tests__/MyLocationScreen.tests.tsx:28]: you don't need the extra library, you can use something like : 
`const item = root.findByProps({ title: ElectricityType.ireland.toUpperCase() });`
and then : 
`item.props.onPress();`
See `SocialMedia.test.tx` as reference

**Review by PierreBresson — CHANGES_REQUESTED** [review]

**Comment by PierreBresson** [comment]
Closing for inactivity

**Pull request closed without merging**